hopsanc/src/hopsanc.cpp

```diff
@@ -0,0 +1,275 @@
+            return nullptr;
+        }
+    }
+
+    //Find component
+    hopsan::Component *pComp = pSystem->getSubComponent(splitVar[0]);
+    if(!pComp) {
+        std::cout << "Error: No such component: " << splitVar[0].c_str() << "\n";
+        size = -1;
+        return nullptr;
+    }
+
+    //Find port
+    hopsan::Port *pPort = pComp->getPort(splitVar[1]);
+    if(!pPort) {
+        std::cout << "Error: No such port: " << splitVar[1].c_str() << "\n";
+        size = -1;
+        return nullptr;
+    }
+
+    int varId = pPort->getNodeDataIdFromName(splitVar[2]);
+    if(varId < 0) {
+        std::cout << "Error: No such variable: " << splitVar[2].c_str() << "\n";
+        size = -1;
+        return nullptr;
+    }
+
+    size = int(spCoreComponentSystem->getNumActuallyLoggedSamples());
+    double *data = new double[size];
+    std::vector< std::vector<double> > *pLogData = pPort->getLogDataVectorPtr();
+    for (size_t t=0; t<spCoreComponentSystem->getNumActuallyLoggedSamples(); ++t) {
+        data[t] = (*pLogData)[t][size_t(varId)];
+    }
+
+    return data;
+}
+
+int loadLibrary(const char *path)
+{
+    if(!gHopsanCore.loadExternalComponentLib(path)) {
+        printWaitingMessages(gHopsanCore, false, false);
+        return -1;
+    };
+    printWaitingMessages(gHopsanCore, false, false);
+    return 0;
+}
+
+int setStartTime(double value)
+{
+    startTime = value;
+    return 0;
+}
+
+int setTimeStep(double value)
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded!\n";
+        return -1;
+    }
+    spCoreComponentSystem->setDesiredTimestep(value);
+    return 0;
+}
+
+int setStopTime(double value)
+{
+    stopTime = value;
+    return 0;
+}
+
+int simulate()
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded!\n";
+        return -1;
+    }
+    std::cout << "Checking model... ";
+    if (spCoreComponentSystem->checkModelBeforeSimulation()) {
+        std::cout << "Success!\n";
+    }
+    else {
+        std::cout << "Failed!\n";
+        printWaitingMessages(gHopsanCore, false, false);
+        return -1;
+    }
+
+    std::cout << "Initializing model... ";
+    if(spCoreComponentSystem->initialize(startTime, stopTime)) {
+        std::cout << "Success!\n";
+    }
+    else {
+        std::cout << "Failed!\n";
+        printWaitingMessages(gHopsanCore, false, false);
+        return -1;
+    }
+    printWaitingMessages(gHopsanCore, false, false);
+
+    std::cout << "Simulating... ";
+    spCoreComponentSystem->simulate(stopTime);
+    std::cout << "Finished!\n";
+
+    printWaitingMessages(gHopsanCore, false, false);
+    return 0;
+}
+
+double *getTimeVector(int &size)
+{
+    if(!spCoreComponentSystem) {
+        std::cout << "Error: No model is loaded.\n";
+        size = -1;
+        return nullptr;
+    }
+    size = int(spCoreComponentSystem->getNumActuallyLoggedSamples());
+    return spCoreComponentSystem->getLogTimeVector()->data();
```

Comment:
In this case, returning a pointer to data is OK, since data lifetime is managed inside hopsan core.
But It might be a good idea to let the caller send in a pointer to buffer and size and copy the data, so that the caller owns the allocated memory, incase the caller would decide to store the pointer for later use. 

Or we write in the function documentation that the data must be consumed immediately, because it is no guarantee that the memory will remain valid. Then the caller can clone the data themselves.

Reply:
I modified this function so it works the same way as for `getDataVector`; the array must be pre-buffed by the caller. I skipped the size argument, since it should always equal number of samples.